l1-contracts/deploy-scripts/PrepareZKChainRegistrationCalldata.s.sol

```diff
@@ -0,0 +1,286 @@
+
+        console.log("Computed L2 bridge impl address:", implContractAddress);
+        console.log("Bridge bytecode hash:");
+        console.logBytes32(bridgeBytecodeHash);
+        console.log("Bridge constructor data:");
+        console.logBytes(bridgeConstructorData);
+        console.log("Sender:", msg.sender);
+
+        bytes32 l2StandardErc20BytecodeHash = L2ContractHelper.hashL2Bytecode(bytecodes.beaconProxy);
+
+        // solhint-disable-next-line func-named-parameters
+        bytes memory proxyInitializationParams = abi.encodeWithSignature(
+            "initialize(address,address,bytes32,address)",
+            config.l1SharedBridgeProxy,
+            config.erc20BridgeProxy,
+            l2StandardErc20BytecodeHash,
+            l2GovernorAddress
+        );
+
+        bytes memory l2SharedBridgeProxyConstructorData = abi.encode(
+            implContractAddress,
+            l2GovernorAddress,
+            proxyInitializationParams
+        );
+
+        address proxyContractAddress = L2ContractHelper.computeCreate2Address(
+            msg.sender,
+            salt,
+            L2ContractHelper.hashL2Bytecode(bytecodes.l2SharedBridgeProxyBytecode),
+            keccak256(l2SharedBridgeProxyConstructorData)
+        );
+
+        console.log("Computed L2 bridge proxy address:", proxyContractAddress);
+        console.log("L1 shared bridge proxy:", config.l1SharedBridgeProxy);
+        console.log("L1 ERC20 bridge proxy:", config.erc20BridgeProxy);
+        console.log("L2 governor addr:", l2GovernorAddress);
+
+        return proxyContractAddress;
+    }
+
+    function prepareRegisterHyperchainCall() internal view returns (IGovernance.Call memory) {
+        Bridgehub bridgehub = Bridgehub(config.bridgehub);
+
+        bytes memory data = abi.encodeCall(
+            bridgehub.createNewChain,
+            (
+                config.chainId,
+                config.stateTransitionProxy,
+                config.baseToken,
+                config.bridgehubCreateNewChainSalt,
+                config.chainAdmin,
+                config.diamondCutData
+            )
+        );
+
+        return IGovernance.Call({target: config.bridgehub, value: 0, data: data});
+    }
+
+    function prepareInitializeChainGovernanceCall(
+        address l2SharedBridgeProxy
+    ) internal view returns (IGovernance.Call memory) {
+        L1SharedBridge bridge = L1SharedBridge(config.l1SharedBridgeProxy);
+
+        bytes memory data = abi.encodeCall(bridge.initializeChainGovernance, (config.chainId, l2SharedBridgeProxy));
+
+        return IGovernance.Call({target: config.l1SharedBridgeProxy, value: 0, data: data});
+    }
+
+    function scheduleTransparentCalldata(
+        IGovernance.Call[] memory calls,
+        IGovernance.Call memory initChainGovCall
+    ) internal {
+        IGovernance governance = IGovernance(config.governance);
+
+        IGovernance.Operation memory operation = IGovernance.Operation({
+            calls: calls,
```

Comment:
could all 3 calls be run within a single Operation? 
(if not - please add comment)

Reply:
they could, but there is a security concern, I covered it in a big comment in the top